h5grove/utils.py

```diff
@@ -175,6 +180,28 @@ def parse_bool_arg(query_arg: Union[str, None], fallback: bool) -> bool:
     return query_arg.lower() != "false"
 
 
+def parse_link_resolution_arg(
+    raw_query_arg: Union[str, None], fallback: LinkResolution
+) -> LinkResolution:
+    if raw_query_arg is None:
+        return fallback
+
+    query_arg = raw_query_arg.lower()
+
+    if query_arg == "true" or query_arg == LinkResolution.ALL.value:
+        return LinkResolution.ALL
+
+    if query_arg == "false" or query_arg == LinkResolution.NONE.value:
+        return LinkResolution.NONE
+
+    if query_arg == LinkResolution.ONLY_VALID.value:
```

Comment:
` only_valid == LinkResolution.ONLY_VALID` returns `False` so the check must be done against the value. Basically, each time we need the string value of the enum, I use `.value`.

Using an enum makes the code longer but is more robust to changes in my opinion.

Reply:
I agree!
According to the [`enum` doc](https://docs.python.org/3/library/enum.html?highlight=enum#others), using `str` as a mixing class makes equality `'only_valid' == LinkResolution.ONLY_VALID` works:

```python
class LinkResolution(str, Enum):
    NONE = 'none'
    [...]
```

Reply:
I read this part and thought I could make it work but I got errors.

Reading your comment, I now realize that I wrote `(Enum, str)` instead of `(str, Enum)` :facepalm: ... Thanks for this !